src/lib/libembind.js

```diff
@@ -318,59 +318,50 @@ var LibraryEmbind = {
     }
   },
 
+#if ASSERTIONS
+  $assertIntegerRange: (name, value, minRange, maxRange) => {
+    if (value < minRange || value > maxRange) {
+      throw new TypeError(`Passing a number "${embindRepr(value)}" from JS side to C/C++ side to an argument of type "${name}", which is outside the valid range [${minRange}, ${maxRange}]!`);
+    }
+  },
+#endif
+
   _embind_register_integer__docs: '/** @suppress {globalThis} */',
   // When converting a number from JS to C++ side, the valid range of the number is
   // [minRange, maxRange], inclusive.
   _embind_register_integer__deps: [
     '$integerReadValueFromPointer', '$readLatin1String', '$registerType',
 #if ASSERTIONS
     '$embindRepr',
+    '$assertIntegerRange',
 #endif
   ],
   _embind_register_integer: (primitiveType, name, size, minRange, maxRange) => {
     name = readLatin1String(name);
-    // LLVM doesn't have signed and unsigned 32-bit types, so u32 literals come
-    // out as 'i32 -1'. Always treat those as max u32.
-    if (maxRange === -1) {
-      maxRange = 4294967295;
-    }
 
-    var fromWireType = (value) => value;
+    const isUnsignedType = maxRange < minRange;
```

Comment:
Does this work for all the integer types?  I guess so if the tests pass.

Perhaps comment here.  Something like "When minRange is passed as a negative value from the native side it will show up here is a larger unsigned value.".  

Also, is the optimization of `isUnsignedType` something that is separable perhaps?  Seems like it could land in isolation.. but maybe not worth it?

Reply:
It's not (just) an optimisation, it actually doesn't work correctly sometimes since it relied on name of the type passed from C++ side and I changed how types are registered somewhat. This check is more reliable, and happens to be more optimized as well.

Reply:
> When minRange is passed as a negative value from the native side it will show up here is a larger unsigned value.

FWIW it's the other way around. I'll add a comment.

Reply:
Actually I can simplify this to just `minRange === 0` and `minRange === 0n` correspondingly.

Reply:
Sweeeet!